online-store/src/components/ProductsCarts.ts

```diff
@@ -0,0 +1,29 @@
+import { isJSDocThisTag } from 'typescript';
+import { AppState } from '../Store/AppState';
+import '../styles/productsCarts.css';
+import { ProductCart } from './ProductCart';
+import { IComponent } from './interfaces';
+
+class ProductsCarts implements IComponent {
+  private carts: ProductCart[] = [];
+  constructor() {
+    this.carts = AppState.instance.state.basket.orders.map((item, index) => new ProductCart(index, item.legoItem));
+  }
+  async render() {
+    if (AppState.countProducts() === 0) return ``;
+    else
+      return `
+      <div class="basket-design__products-design products-design">

```

Comment:
давайте везде придерживаться одного стиля - если где-то ставим скобки, то везде ставим)
```suggestion
    if (AppState.countProducts() === 0) {
       return ``;
    } else {
      return `
      <div class="basket-design__products-design products-design">

```